Extending/FileSystemProviders/index.md

```diff
@@ -1,48 +1,159 @@
+        }
+    }
+}
+```
+The PhysicalFileSystem accepts the application path to the 'virtual Root' to location the Media folder the ~ is important!
+
+Alternatively you can combine:
+a **RootPath** - the rooted, filesystem path, using directory separator chars, NOT ending with a separator // eg "c:" or "c:\path\to\site" or "\\server\path"
+and **RootUrl** - the relative url, using url separator chars, NOT ending with a separator // eg "" or "/Views" or "/Media" or "/<vpath>/Media" in case of a virtual path
+
+:::note
+In Umbraco V7, this configuration was located a physical file: `/config/FileSystemProviders.config`.
+:::
 
 `PhysicalFileSystem` implements the `IFileSystem` interface, and it is possible to replace it with a custom class - eg. if you want your media files stored on Azure or something similar.
 
 If you configure Umbraco to use a custom file system provider for media, you most likely won't need to access the implementation directly. Umbraco uses a wrapper class called `MediaFileSystem`. You can get a reference to this wrapper class with the following code:
 
 ```csharp
-MediaFileSystem media = FileSystemProviderManager.Current.GetFileSystemProvider<MediaFileSystem>();
+IMediaFileSystem media = Current.MediaFileSystem;
+```
+or via dependency injection in the constructor for your custom class or controller:
+```csharp
+ public class ImagesController : UmbracoAuthorizedApiController
+    {
+        private readonly IMediaFileSystem _mediaFileSystem;
+
+
+        public ImagesController(IMediaFileSystem mediaFileSystem)
+        {
+            _mediaFileSystem = mediaFileSystem;
+        }
 ```
 
 This will be enough in most cases and is the way Umbraco will access the file system provider. The wrapper class implements the same interface `IFileSystem` as any custom providers should do, so you will be able to call the same methods.
 
-If your custom file system provider has some extra logic that you need to access, you can of course get a reference to the actual instance of IFileSystem. This can be archived through the following code:
+## Setting the PhysicalFileSystem to be outside of the application root
 
-```csharp
-IFileSystem fs = FileSystemProviderManager.Current.GetUnderlyingFileSystemProvider("media");
+## MediaPath Scheme
+
+The MediaPath Scheme defines the current set of rules that decide the format of the Media Path when it is saved into the media archive wherever it is located.
+
+By default the MediaPath scheme used by Umbraco is the 'UniqueMediaPathScheme' this generates a unique 'folder' to place the uploaded image in eg
+
+/media/dozdrg2f/mylovelyimage.jpg
+
+'/media' is defined by the PhysicalFileSystem and 'dozdrg2f' is generated by the UniqueMediaPathScheme.
+:::note
+In Umbraco 7 the integer ids were used in the path, and this approach is still possible using the 'OriginalMediaPathScheme'
+:::
+are stored in the format ~/media/
```

Comment:
Is this sentence supposed to be inside the note? 🤔

Reply:
just shouldn't be there :-)